packages/yoastseo/src/languageProcessing/researches/matchKeywordInSubheadings.js

```diff
@@ -30,25 +45,32 @@ const numberOfSubheadingsReflectingTopic = function( topicForms, subheadings, us
 /**
  * Checks if there are any h2 or h3 subheadings in the text and if they have the keyphrase or synonyms in them.
  *
- * @param {Object}     paper      The paper object containing the text and keyword.
- * @param {Researcher} researcher The researcher object.
+ * @param {Paper}		paper      The paper object containing the text and keyword.
+ * @param {Researcher}	researcher The researcher object.
  *
- * @returns {Object} The result object.
+ * @returns {KeyphraseInSubheadingsResult} An object containing the number of subheadings,
+ * the number of subheadings reflecting the topic, the percentage of subheadings reflecting the topic and an empty string.
  */
 export default function matchKeywordInSubheadings( paper, researcher ) {
 	const functionWords = researcher.getConfig( "functionWords" );
 
 	// A custom helper to match word in text.
 	const matchWordCustomHelper = researcher.getHelper( "matchWordCustomHelper" );
-
+	const customCountLength = researcher.getHelper( "customCountLength" );
 
 	let text = paper.getText();
 	text = removeHtmlBlocks( text );
 	text = filterShortcodesFromHTML( text, paper._attributes && paper._attributes.shortcodes );
 	text = stripSomeTags( text );
 	const topicForms = researcher.getResearch( "morphology" );
 	const locale = paper.getLocale();
-	const result = { count: 0, matches: 0, percentReflectingTopic: 0 };
+	const result = {
+		count: 0,
+		matches: 0,
+		percentReflectingTopic: 0,
+		text: text,
+		textLength: customCountLength ? customCountLength( text ) : getWords( text ).length,
```

Comment:
Also return the text length here so we don't need to recalculate it inside the assessment